documentation/api/io.strimzi.api.kafka.model.KafkaAuthorizationCustom.adoc

```diff
@@ -0,0 +1,131 @@
+config:
+  strimzi.authorization.custom.property1=value1
+  strimzi.authorization.custom.property2=value2
+----
+
+A sample `org.apache.kafka.server.authorizer.Authorizer` implementation
+
+.An example of sample custom authorizer implementation in Java
+[source,java]
+----
+public class SampleAuthorizer implements org.apache.kafka.server.authorizer.Authorizer {
+    private List<UserSpec> superUsers = Collections.emptyList();
+
+    @Override
+    public void configure(Map<String, ?> configs) {
+        // configure super users
+        String users = (String) configs.get("super.users");
+        if (users != null) {
+            superUsers = Arrays.asList(users.split(";"))
+                    .stream()
+                    .map(s -> UserSpec.of(s))
+                    .collect(Collectors.toList());
+        }
+    }
+
+    @Override
+    public List<AuthorizationResult> authorize(AuthorizableRequestContext requestContext, List<Action> actions) {
+
+        if (isSuperUser(requestContext.principal())) {
+            // allow all the operations to make sure Strimzi works correctly
+        }
+
+        // TODO: your custom implementation
+        return null;
+    }
+
+    private boolean isSuperUser(KafkaPrincipal principal) {
+        for (UserSpec u : superUsers) {
+            if (principal.getPrincipalType().equals(u.getType())
+                    && principal.getName().equals(u.getName())) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    @Override
+    public List<? extends CompletionStage<AclCreateResult>> createAcls(AuthorizableRequestContext requestContext,
+            List<AclBinding> aclBindings) {
+        // TODO: your custom implementation
+    }
+
+    @Override
+    public List<? extends CompletionStage<AclDeleteResult>> deleteAcls(AuthorizableRequestContext requestContext,
+            List<AclBindingFilter> aclBindingFilters) {
+        // TODO: your custom implementation
+    }
+
+    @Override
+    public Iterable<AclBinding> acls(AclBindingFilter filter) {
+        // TODO: your custom implementation
+    }
+
+    public static class UserSpec {
+        private final String type;
+        private final String name;
+
+        private UserSpec(String type, String name) {
+            this.type = type;
+            this.name = name;
+        }
+
+        public String getType() {
+            return type;
+        }
+
+        public String getName() {
+            return name;
+        }
+
+        public static UserSpec of(String principal) {
+            int pos = principal.indexOf(':');
+            if (pos <= 0) {
+                throw new IllegalArgumentException("Invalid user specification: " + principal);
+            }
+            return new UserSpec(principal.substring(0, pos), principal.substring(pos + 1));
+        }
+
+        @Override
+        public String toString() {
+            return super.toString() + " " + type + ":" + name;
+        }
+    }
+}
+----
```

Comment:
I not sure this example makes sense ... the docs would need to be checked and maintained when the Authorizer interface ever changes. Maybe we should leave it to Kafka.

Reply:
Yes, agree. Will remove it